staging/src/k8s.io/apiserver/pkg/server/options/encryptionconfig/types.go

```diff
@@ -69,3 +71,14 @@ type Key struct {
 
 // IdentityConfig is an empty struct to allow identity transformer in provider configuration.
 type IdentityConfig struct{}
+
+// KMS contains the name and path to configuration file for a KMS based envelope transformer.
+type KMSConfig struct {
+	// name is the name of the KMS plugin to be used.
+	Name string `json:"name"`
+	// cacheSize is the maximum number of secrets which are cached in memory. The default value is 1000.
+	// +optional
+	CacheSize int `json:"cachesize,omitempty"`
```

Comment:
It is not specific to the plugin's configuration. It is a property of the envelope transformer. It will also be used by cloudprovidedkms based plugins.